core-concepts/modules/ROOT/pages/typeql/sql-vs-typeql.adoc

```diff
@@ -0,0 +1,718 @@
+----
+SELECT 
+  COUNT(*) as total,
+  AVG(age) as avg_age,
+  MAX(age) as max_age,
+  MIN(age) as min_age,
+  SUM(age) as sum_age
+FROM person;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=1]
+match
+  $p isa person, has age $age;
+reduce
+  $total = count,
+  $avg_age = mean($age),
+  $max_age = max($age),
+  $min_age = min($age),
+  $sum_age = sum($age);
+----
+
+=== HAVING → reduce + match
+
+**SQL:**
+[source,sql]
+----
+SELECT age, COUNT(*) as count
+FROM person
+GROUP BY age
+HAVING COUNT(*) > 1;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=0]
+match
+  $p isa person, has age $age;
+reduce $count = count groupby $age;
+match $count > 1;
+----
+
+== Sorting and Pagination
+
+=== ORDER BY → sort
+
+**SQL:**
+[source,sql]
+----
+SELECT name, age FROM person ORDER BY age DESC, name ASC;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has name $name, has age $age;
+sort $age desc, $name asc;
+select $name, $age;
+----
+
+=== LIMIT and OFFSET
+
+**SQL:**
+[source,sql]
+----
+SELECT name FROM person ORDER BY name LIMIT 10 OFFSET 20;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=0]
+match
+  $p isa person, has name $name;
+sort $name;
+offset 20;
+limit 10;
+select $name;
+----
+
+== Advanced Operations
+
+=== DISTINCT → distinct
+
+**SQL:**
+[source,sql]
+----
+SELECT DISTINCT age FROM person;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has age $age;
+select $age;
+distinct;
+----
+
+=== Subqueries → Functions or Nested match
+
+**SQL:**
+[source,sql]
+----
+SELECT name FROM person
+WHERE age > (SELECT AVG(age) FROM person);
+----
+
+**TypeQL using functions:**
+[,typeql]
+----
+#!test[read, count=1]
+with fun avg_age() -> integer:
+  match $p isa person, has age $age;
+  return mean($age);
+match
+  $p isa person, has name $name, has age $age;
+  $age > avg_age();
+select $name;
+----
+
+=== UNION → or patterns
+
+**SQL:**
+[source,sql]
+----
+SELECT name FROM person WHERE age < 25
+UNION
+SELECT name FROM person WHERE age > 35;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=1]
+match
+  $p isa person, has name $name, has age $age;
+  { $age < 25; } or { $age > 35; };
+select $name;
+----
+
+=== EXISTS → not { not { ... } }
+
+**SQL:**
+[source,sql]
+----
+SELECT name FROM person p
+WHERE EXISTS (
+  SELECT 1 FROM employment e
+  WHERE e.employee_id = p.id OR e.employer_id = p.id
+);
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=2]
+match
+  $p isa person, has name $name;
+  not { not { $e isa employment, links (employee: $p); }; };
+select $name;
+----
+
+_This will be improved in future versions of TypeQL!_
+
+=== IN → Multiple value constraints
+
+**SQL:**
+[source,sql]
+----
+SELECT name FROM person WHERE age IN (25, 30, 35);
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=1]
+match
+  $p isa person, has name $name, has age $age;
+  { $age == 25; } or { $age == 30; } or { $age == 35; };
```

Comment:
Aside: once we have lists, we should be able to say `$age in [25, 30, 35]`. Note that `$age` is an attribute while the list contains raw value, we should handle that.